contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault} from "../interfaces/IRibbon.sol";
+import {IRibbonThetaVault} from "../interfaces/IRibbonThetaVault.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+interface IVaultQueue {
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        uint256 amount;
+        uint256 timestamp;
+    }
+}
+
+contract VaultQueue is ReentrancyGuardUpgradeable, Ownable {
+    mapping(address => IVaultQueue.Transfer[]) public qTransfer;
+    mapping(address => uint256) public totalAmount;
+    address[] public vaults;
+
+    function interVaultTransfer() external payable onlyOwner nonReentrant {
+        for (uint256 i = 0; i < vaults.length; i++) {
+            IRibbonVault(vaults[i]).completeWithdraw();
```

Comment:
Since it's likely that not *all* vaults will have withdrawals, this needs to check if there is an open withdrawal for the vault before calling `completeWithdraw`

Reply:
This is checked at the beginning of `_completeWithdraw()` and it seems like it adds more gas to check, since it would be checked again regardless.

Reply:
Done